### ISSUE TITLE
Psalm should infer that result of multiplication is never an array.

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#2123]
The same goes for all numeric operations, excluding addition.

This example is used to show the types psalm infers

```php
<?php
$x = getopt('', ['value:']);
// correctly infers array<int, mixed>|string|false for $x['value']
// Observed:  PossiblyInvalidOperand - src/array.php:5:23 - Cannot add an array to a non-array int
// Expected:  PossiblyInvalidOperand - src/array.php:5:23 - Cannot use an array and a non-array int in a binary operation (or something along those lines)
$result = $x['value'] * 3;
// Observed: Argument 1 of fread expects resource, array<array-key, mixed> provided
// Expected: Should warn about $result being numeric
fread($result, 1);
```

**Comment by orklah** [comment]
https://psalm.dev/r/8a2d4ba8b7

**Comment by kkmuffme** [comment]
Afaik this is sufficiently handled by giving the PossiblyInvalidOperand error, which requires you to take action.
Fixing this would not really improve psalm, as you have an error in your code and psalm reports it.

**Comment by kkmuffme** [comment]
Close the issue?

**Comment by orklah** [comment]
No, as I said, while it's not a big priority, there's issue with that who should be fixed one day

**Comment by robchett** [comment]
Under my changes these lines will no longer trigger additional errors as `$result` will be of type `int|float`

```php
$result2 = $result + 1;
$result3 = $result * 2;
```
